robottelo/api/utils.py

```diff
@@ -589,75 +603,36 @@ def configure_puppet_test():
         organization=org,
         name=gen_string('alpha')
     ).create()
-
-    # step 2: Clone and Upload manifest
+    # Clone and Upload manifest
     with manifests.clone() as manifest:
         upload_manifest(org.id, manifest.content)
-
-    # step 3: Sync RedHat Sattools RHEL6 and RHEL7 repository
-    repos = [
-        entities.Repository(id=enable_rhrepo_and_fetchid(
-            basearch='x86_64',
-            org_id=org.id,
-            product=PRDS['rhel'],
-            repo=value['repo'],
-            reposet=value['reposet'],
-            releasever=None,
-        ))
-        for value in repo_values
-        ]
-    for repo in repos:
-        repo.sync()
-
-    # step 4: Create content view
+    # Create content view
     content_view = entities.ContentView(
         organization=org,
         name=gen_string('alpha')
     ).create()
-
-    # step 5: Associate repository to new content view
-    content_view.repository = repos
-    content_view = content_view.update(['repository'])
-
-    # step 6: Publish content view and promote to lifecycle env.
-    content_view.publish()
-    content_view = content_view.read()
-    promote(content_view.version[0], env.id)
-
-    # step 7: Create activation key
-    ak_name = gen_string('alpha')
-    activation_key = entities.ActivationKey(
-        name=ak_name,
-        environment=env,
-        organization=org,
-        content_view=content_view,
-    ).create()
-
-    # step 7.1: Walk through the list of subscriptions.
-    # Find the "Employee SKU" and attach it to the
-    # recently-created activation key.
-    for sub in entities.Subscription(organization=org).search():
-        if sub.read_json()['product_name'] == DEFAULT_SUBSCRIPTION_NAME:
-            # 'quantity' must be 1, not subscription['quantity']. Greater
-            # values produce this error: "RuntimeError: Error: Only pools
-            # with multi-entitlement product subscriptions can be added to
-            # the activation key with a quantity greater than one."
-            activation_key.add_subscriptions(data={
-                'quantity': 1,
-                'subscription_id': sub.id,
-            })
-            break
-    for content_label in [REPOS['rhst6']['id'], REPOS['rhst7']['id']]:
-        # step 7.2: Enable product content
-        activation_key.content_override(data={'content_override': {
-            u'content_label': content_label,
-            u'value': u'1',
-        }})
+    # Create two activation keys for rhel7 and rhel6
+    for repo in repo_values:
+        activation_key = entities.ActivationKey(
+            name=repo.get('akname'),
+            environment=env,
+            organization=org,
+        ).create()
+        # Setup org for a RH or custom repo for RHEL6 and RHEL7
+        setup_org_for_a_rh_repo({
```

Comment:
why you are using the CLI factory function in API, can't we use API equivalent functions?

Reply:
@ldjebran because , Manifest upload in CLI doesn't timeout. This is the issue for upgrade boxes , when trying to upload. This is just a helper can be used in CLI, API,UI . I can move this to CLI factory helpers if this isn't the right place for this.

Reply:
@ldjebran @renzon on second thought , would it make sense to have longrun helpers, long run tests are meant to test the functionality of end-to-end scenarios , so it doesn't matter where the helpers would come from.

Reply:
In my opinion, we should keep API and CLI factory helpers using the pure relevant interface, if  we need some fast hybrid functions perhaps we should move that code to the test module that use it, that way we keep interfaces factories clean.

Reply:
Moving it to test module ,as it will be only used by oscap